src/ResourceManager/CognitiveServices/Commands.Management.CognitiveServices/CognitiveServicesAccount/GetAzureCognitiveServicesAccountSkus.cs

```diff
@@ -15,45 +15,90 @@
-using CognitiveServicesModels = Microsoft.Azure.Commands.Management.CognitiveServices.Models;
 
 namespace Microsoft.Azure.Commands.Management.CognitiveServices
 {
     /// <summary>
     /// Get Available Skus for Cognitive Services Account
     /// </summary>
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesAccountSkus"), OutputType(typeof(CognitiveServicesModels.PSCognitiveServicesSkus))]
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesAccountSkus", DefaultParameterSetName = GetSkusWithAccountParamSetName),
+        OutputType(typeof(PSCognitiveServicesSkus), typeof(ResourceSku))]
     public class GetAzureCognitiveServicesAccountSkusCommand : CognitiveServicesAccountBaseCmdlet
     {
+        protected const string GetSkusWithAccountParamSetName = "GetSkusWithAccount";
+        protected const string GetSkusWithFilterParamSetName = "GetSkusWithFilter";
+
+        [Parameter(
+            ParameterSetName = GetSkusWithFilterParamSetName,
+            Position = 0,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Type.")]
+        [Alias(CognitiveServicesAccountTypeAlias, AccountTypeAlias, KindAlias)]
+        public string Type { get; set; }
+
+        [Parameter(
+            ParameterSetName = GetSkusWithFilterParamSetName,
+            Position = 1,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Location.")]
+        [LocationCompleter("Microsoft.CognitiveServices/accounts")]
+        public string Location { get; set; }
+
         [Parameter(
+            ParameterSetName = GetSkusWithAccountParamSetName,
             Position = 0,
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Resource Group Name.")]
         [ResourceGroupCompleter()]
-        [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
         [Parameter(
+            ParameterSetName = GetSkusWithAccountParamSetName,
             Position = 1,
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Cognitive Services Account Name.")]
         [Alias(CognitiveServicesAccountNameAlias, AccountNameAlias)]
-        [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         public override void ExecuteCmdlet()
         {
             base.ExecuteCmdlet();
             RunCmdLet(() =>
             {
-                var cognitiveServicesKeys = this.CognitiveServicesClient.Accounts.ListSkus(
-                     this.ResourceGroupName,
-                     this.Name);
+                if (!string.IsNullOrEmpty(this.Name))
+                {
+                    if (string.IsNullOrEmpty(this.ResourceGroupName))
```

Comment:
@yangyuan this check shouldn't be needed because PowerShell will prompt the user to provide a value for the `-ResourceGroupName` parameter if a value is provided for the `-Name` parameter

Reply:
Ya, you are right.

Reply:
I actually changed to use switch(ParameterSetName), and use original code in legacy ParameterSet